Fix confusing wording in Asset Pipeline guide [ci skip]

rmacklin opened the pull request:
### Summary

While looking into https://github.com/rails/rails/issues/38307, I noticed some confusing wording in the Asset Pipeline guide:
https://github.com/rails/rails/blob/aeac447b0ca750294fc0556bd0b9842ecce5ad0f/guides/source/asset_pipeline.md#L36-L45

Specifically
> adding _them_ to your `Gemfile`

and
> add _those_ gems

is confusing because the preceding paragraph only mentions _one_ gem (`sass-rails`). I discovered that this is a result of removing mentions of `uglifier` and `coffee-rails` from that paragraph in https://github.com/rails/rails/pull/35994.

So I've opened this PR as a follow-up to https://github.com/rails/rails/pull/35994 to clean up the confusing wording that was left in the guide.

### Other Information

While I was in there, I also clarified the purpose of the `sass-rails` gem and added links to https://github.com/rails/sass-rails and https://sass-lang.com.

rmacklin committed:
Clean up leftover plural references in Asset Pipeline guide

Prior to ab123a33d2460dcdc5c36001cef5316eadc75fb3, this guide mentioned
that three gems (sass-rails, uglifier, and coffee-rails) would be added
to the Gemfile by default, unless the `--skip-sprockets` option was
used when generating the application.
However, ab123a33d2460dcdc5c36001cef5316eadc75fb3 removed uglifier and
coffee-rails from the guide (to reflect the switch to webpacker for
managing javascript in rails 6). Now that only sass-rails is left, the
guide should say "adding this to your Gemfile" instead of "adding them
to your Gemfile" and "add that gem" instead of "add those gems".

[ci skip]

rmacklin committed:
Clarify what sass-rails is used for in Asset Pipeline guide

The original text "used by Sprockets for asset compression" was written
when this sentence was summarizing the purpose of three gems:
sass-rails, uglifier, and coffee-rails.
However, ab123a33d2460dcdc5c36001cef5316eadc75fb3 removed uglifier and
coffee-rails from this sentence. Now that only sass-rails is left, we
can be more specific about what it is used for: compiling Sass
stylesheets.

[ci skip]

rmacklin committed:
Link to the sass-rails repo and sass-lang.com in Asset Pipeline guide

We link to the sprockets-rails repository in the preceding paragraph,
so it seems appropriate to also link to the sass-rails repository in
this paragraph. And since readers of this guide may not know what Sass
is, it also seems appropriate to link to the sass-lang.com website.

[ci skip]

eugeneius commented on guides/source/asset_pipeline.md:
```diff
@@ -33,16 +33,17 @@ passing the `--skip-sprockets` option.
-them to your `Gemfile`, so if you later want to enable
-the asset pipeline you will have to add those gems to your `Gemfile`. Also,
+this to your `Gemfile`, so if you later want to enable
+the asset pipeline you will have to add that gem to your `Gemfile`. Also,
```
> Rails automatically **adds the `sass-rails` gem to your `Gemfile`** [...] Using the `--skip-sprockets` option will prevent Rails from **adding this to your `Gemfile`**, so if you later want to enable the asset pipeline you will have to **add that gem to your `Gemfile`**.

This reminds me of:

["By installing Java, you will be able to experience the power of Java."](https://www.joelonsoftware.com/2009/01/12/by-installing-java-you-will-be-able-to-experience-the-power-of-java/) 😄 

Perhaps:

```suggestion
this gem, so if you later want to enable the asset pipeline
you will have to add it to your `Gemfile` manually. Also,
```

rmacklin replied:
👍 I've applied your suggestion in fd9a032eea94371cd811c149f3b5f0e64ec0ee69

eugeneius reviewed (commented):
There are a few more outdated references in the [Preprocessing section](https://edgeguides.rubyonrails.org/asset_pipeline.html#preprocessing), in particular the assertion that Rails generates a CoffeeScript file with every scaffold by default. Could you do a pass over those too?

rmacklin committed:
Further reword Asset Pipeline guide introduction to be less repetitive

[ci skip]

Co-Authored-By: Eugene Kenny <elkenny@gmail.com>

rmacklin committed:
Remove mention of generating CoffeeScript files in Preprocessing section

of the Asset Pipeline guide

This section previously declared that the Rails controller generator
generated an `app/assets/javascripts/projects.coffee` and an
`app/assets/stylesheets/projects.scss` file. Then, in
ab123a33d2460dcdc5c36001cef5316eadc75fb3, we removed the assertion that
`app/assets/javascripts/projects.coffee` would be generated (to reflect
the switch to webpacker for managing javascript in Rails 6). However,
the preceding sentence and the subsequent paragraph were still written
with the assumption that a CoffeeScript file would be generated.

Now we've reworded those parts to be consistent with the fact that Rails
no longer generates a CoffeeScript file.

[ci skip]

rmacklin commented:
Sounds good, I've updated the Preprocessing section in e71692f3f5ef4a35f3213cbc8d6d6095ab8791aa

eugeneius commented on guides/source/asset_pipeline.md:
```diff
@@ -509,17 +510,16 @@ might concatenate three CSS files together this way:
+`app/assets/stylesheets/projects.scss` file.
+
+In development mode, or if the asset pipeline is disabled, when this file is
+requested it is processed by the processor provided by the `sass` gem and then
```
The `sass` gem is no longer maintained, and `sass-rails` now depends on the `sassc` gem instead via `sassc-rails` (https://github.com/rails/rails/issues/32896).

Even if it's not technically accurate, I think this should say "provided by the `sass-rails` gem" - for the purposes of following this guide, it's enough to know that the processor is available because `sass-rails` is in the Gemfile, even if the code is ultimately provided by a different gem.

rmacklin replied:
👍 Updated to `sass-rails` in 649d265e16b8333f5a86aa3de46ae4aac6635e30

rmacklin committed:
Remove mention of the deprecated `sass` gem from Asset Pipeline guide

As of https://github.com/rails/sass-rails/commit/409d871c2a971691eac86c5a32e614e2941c72b8
sass-rails no longer uses the `sass` gem (which is no longer
maintained; it has been replaced by the `sassc` gem). Thus, the rails
guides should not mention the `sass` gem specifically. Instead, they can
continue to reference the `sass-rails` gem.

[ci skip]

eugeneius commented:
Could you squash your commits? Otherwise I think this is ready to go.

rmacklin commented:
Can you do a squash-and-merge? Or do you want me to squash into more than 1 commit?

eugeneius commented:
If I use the squash and merge button, the autogenerated commit message will be enormous:

```
Fix confusing wording in Asset Pipeline guide [ci skip]

* Clean up leftover plural references in Asset Pipeline guide

Prior to ab123a33d2460dcdc5c36001cef5316eadc75fb3, this guide mentioned
that three gems (sass-rails, uglifier, and coffee-rails) would be added
to the Gemfile by default, unless the `--skip-sprockets` option was
used when generating the application.
However, ab123a33d2460dcdc5c36001cef5316eadc75fb3 removed uglifier and
coffee-rails from the guide (to reflect the switch to webpacker for
managing javascript in rails 6). Now that only sass-rails is left, the
guide should say "adding this to your Gemfile" instead of "adding them
to your Gemfile" and "add that gem" instead of "add those gems".

[ci skip]

* Clarify what sass-rails is used for in Asset Pipeline guide

The original text "used by Sprockets for asset compression" was written
when this sentence was summarizing the purpose of three gems:
sass-rails, uglifier, and coffee-rails.
However, ab123a33d2460dcdc5c36001cef5316eadc75fb3 removed uglifier and
coffee-rails from this sentence. Now that only sass-rails is left, we
can be more specific about what it is used for: compiling Sass
stylesheets.

[ci skip]

* Link to the sass-rails repo and sass-lang.com in Asset Pipeline guide

We link to the sprockets-rails repository in the preceding paragraph,
so it seems appropriate to also link to the sass-rails repository in
this paragraph. And since readers of this guide may not know what Sass
is, it also seems appropriate to link to the sass-lang.com website.

[ci skip]

* Further reword Asset Pipeline guide introduction to be less repetitive

[ci skip]

Co-Authored-By: Eugene Kenny <elkenny@gmail.com>

* Remove mention of generating CoffeeScript files in Preprocessing section

of the Asset Pipeline guide

This section previously declared that the Rails controller generator
generated an `app/assets/javascripts/projects.coffee` and an
`app/assets/stylesheets/projects.scss` file. Then, in
ab123a33d2460dcdc5c36001cef5316eadc75fb3, we removed the assertion that
`app/assets/javascripts/projects.coffee` would be generated (to reflect
the switch to webpacker for managing javascript in Rails 6). However,
the preceding sentence and the subsequent paragraph were still written
with the assumption that a CoffeeScript file would be generated.

Now we've reworded those parts to be consistent with the fact that Rails
no longer generates a CoffeeScript file.

[ci skip]

* Remove mention of the deprecated `sass` gem from Asset Pipeline guide

As of https://github.com/rails/sass-rails/commit/409d871c2a971691eac86c5a32e614e2941c72b8
sass-rails no longer uses the `sass` gem (which is no longer
maintained; it has been replaced by the `sassc` gem). Thus, the rails
guides should not mention the `sass` gem specifically. Instead, they can
continue to reference the `sass-rails` gem.

[ci skip]

Co-authored-by: Eugene Kenny <elkenny@gmail.com>
```

This is way too much detail for the patch it describes. I could edit it down myself, but I'd prefer if you handled it by squashing the commits and writing a shorter combined commit message.

rmacklin commented:
I can squash and push later when I return to my rails dev machine. (Unfortunately, recloning on my current connection is not feasible.)

Alternatively, if you want to merge now, here's a proposal for an abbreviated commit message:
```
Fix confusing wording in Asset Pipeline guide

* Clean up leftover plural "gems" references as a follow-up to
  ab123a33d2460dcdc5c36001cef5316eadc75fb3 (#38307)
* Remove mention of generating CoffeeScript files in Preprocessing section
* Clarify what sass-rails is used for
* Link to the sass-rails repo and sass-lang.com
* Remove mention of the deprecated `sass` gem to reflect the change from
  https://github.com/rails/sass-rails/commit/409d871c2a971691eac86c5a32e614e2941c72b8

[ci skip]
```

eugeneius commented:
Perfect, thanks! 💛 I squash-merged with that message.
